src/lib/Folder.interface.ts

```diff
@@ -0,0 +1,401 @@
+}
+
+/**
+ * Holder
+ */
+export enum Holder {
+  APPLICANT = 'APPLICANT',
+  CO_APPLICANT = 'CO_APPLICANT',
+}
+
+/**
+ * Document properties
+ */
+export interface DocumentProperties {
+  accountCoOwner: string;
+  accountOwner: string;
+  address: Address;
+  annualGrossIncomes: number;
+  bankName: BankName;
+  bic: string;
+  birthCity: string;
+  birthCountry: string;
+  birthDate: string;
+  birthDepartment: number;
+  birthDistrict: number;
+  birthName: string;
+  civility: Civility;
+  deliveredAt: string;
+  documentNumber: string;
+  expiredAt: Date;
+  firstName: string;
+  iban: string;
+  lastName: string;
+  maritalStatus: MaritalStatus;
+  mrzLines: string[];
+  nationality: string;
+  netMonthlyIncomes: number;
+  numberOfChildren: number;
+  score: number;
+}
+
+/**
+ * Address
+ */
+export interface Address {
+  additionalInformation: string;
+  city: string;
+  street: string;
+  zipCode: string;
+}
+
+/**
+ * Enumeration of bank names
+ */
+export enum BankName {
+  ALLIANZ = 'ALLIANZ',
+  AXA = 'AXA',
+  BANQUE_POPULAIRE = 'BANQUE_POPULAIRE',
+  BARCLAYS = 'BARCLAYS',
+  BNP = 'BNP',
+  BOURSORAMA = 'BOURSORAMA',
+  CAISSE_DEPOTS_CONSIGNATIONS = 'CAISSE_DEPOTS_CONSIGNATIONS',
+  CAISSE_EPARGNE = 'CAISSE_EPARGNE',
+  CAIXA_GERAL_DEPOSITOS = 'CAIXA_GERAL_DEPOSITOS',
+  NICKEL = 'NICKEL',
+  CNE = 'CNE',
+  CREDIT_AGRICOLE = 'CREDIT_AGRICOLE',
+  CREDIT_NORD = 'CREDIT_NORD',
+  CREDIT_LYONNAIS = 'CREDIT_LYONNAIS',
+  CREDIT_MUNICIPAL = 'CREDIT_MUNICIPAL',
+  CREDIT_MUTUEL = 'CREDIT_MUTUEL',
+  CZAM_CARREFOUR = 'CZAM_CARREFOUR',
+  FORTUNEO = 'FORTUNEO',
+  GROUPAMA = 'GROUPAMA',
+  CIC = 'CIC',
+  HSBC = 'HSBC',
+  ING_DIRECT = 'ING_DIRECT',
+  BANQUE_POSTALE = 'BANQUE_POSTALE',
+  LCL = 'LCL',
+  LB = 'LB',
+  MONABANQ = 'MONABANQ',
+  SOCIETE_GENERALE = 'SOCIETE_GENERALE',
+  STE_MARSEILLAISE_CREDIT = 'STE_MARSEILLAISE_CREDIT',
+  SYGMA_BANQUE = 'SYGMA_BANQUE',
+  CETELEM = 'CETELEM',
+  ONEY = 'ONEY',
+  CREDIT_FONCIER_FRANCE = 'CREDIT_FONCIER_FRANCE',
+  CREDIPAR = 'CREDIPAR',
+  DIAC = 'DIAC',
+  FACET = 'FACET',
+  FIDEM = 'FIDEM',
+  FINANCO = 'FINANCO',
+  FRANFINANCE = 'FRANFINANCE',
+  MY_MONEY_BANK = 'MY_MONEY_BANK',
+  CARREFOUR_BANK = 'CARREFOUR_BANK',
+  SOCRAM = 'SOCRAM',
+  SOFINCO = 'SOFINCO',
+  VIAXEL = 'VIAXEL',
+  OTHER = 'OTHER',
+}
+
+/**
+ * File
+ */
+export interface File {
+  createdAt: Date;
+  id: string;
+  internalComments: InternalComment[];
+  metadata: { extension: string; mimeType: string; name: string; size: number };
+  properties: DocumentProperties;
+  rejectionCode: number;
+  signedUrl: string;
+  signedUrlCreatedAt: Date;
+  signedUrlTTL: number;
+  state: FileState;
+  updatedAt: Date;
+}
+
+/**
+ * Legal file
+ */
+export interface LegalFile extends File {
+  type: LegalFileType;
+}
+
+/**
+ * Supporting file
+ */
+export interface SupportingFile extends File {
+  type: SupportingFileType;
+}
+
+/**
+ * State of the file
+ */
+export enum FileState {
+  IN_PROGRESS = 1,
```

Comment:
It looks weird that `0`  is not used but it is defined like that in the documentation.

Reply:
Yes, because we wanted to be consistent with document status